### ISSUE TITLE
Bugs: minimize/close, auto-lock, new extension [macOS]

### ISSUE THREAD

**Issue by nlydv** [keeweb/keeweb#1820]
### Minimize
The "minimize to taskbar instead of close" feature seems to have broken (disabled?) after the latest updates. With the setting activated, closing the window still doesn't quit the app entirely but the taskbar icon never shows up anymore and the app icon in dock remains.

### Auto-lock
In addition, the auto-lock settings seem to be ignored when "minimized". None of the auto-lock triggers are enabled in the app, yet it always ends up getting locked anyways. Functionally, the effect of "minimizing on close" seems to be the same as actually quitting, except it still shows up in the dock as a running app.

### New extension
Pretty much makes using the new KeeWeb Connect extension pointless unless I always keep it as a visible (or _actually_ minimized) window at all times. I was previously using the keewebhttp plugin with 3rd party browser extension and that doesn't seem to work anymore either. Tried on Safari just to test it, desktop app connected then disconnected, Safari extension page still thinks it's connected.

Will probably end up uninstalling and rebuilding prior version from source.

Still excited to see a dedicated extension with multi-browser support being worked on though!
Unless I'm the only one with these issues, maybe it would've been better to have initially pushed it as a beta release branch?

**Comment by antelle** [comment]
Hi!

Minimize instead of close is somewhat unloved feature, both in Electron, and in KeeWeb. The API is a bit not consistent across operating systems, and not many people are using it, so you may find it not very smooth.

> The "minimize to taskbar instead of close" feature seems to have broken

✅  I can confirm, it behaves like this for me too, will look into it

> auto-lock settings seem to be ignored when "minimized"

🚫  You mean, closed? Because you can't minimize the app on macOS now. When KeeWeb is closed, it's always locked, there's no way to prevent it. So, this works as expected.

> Safari extension page still thinks it's connected.

🚫  Extensions unfortunately communicate events to Safari without a request from the browser. If you refresh the extension options page in Safari, it will show that it's disconnected. Works not as it should do ideally, but expected.

> rebuilding prior version from source.

❗  It will not include KeeWebHttp, this plugin has already been removed from the gallery.

> beta release branch?

There's no beta release for desktop now, not sure if there would be enough audience for a beta version. Maybe yes.

**Comment by nlydv** [comment]
Thanks for the reply @antelle.

> Minimize instead of close ... not many people are using it

Really? That's surprising. From a UI/UX perspective it's one of the reasons that got me to choose KeeWeb after trying all the other clients out there!

> You mean, closed?

Yeah, the auto-lock issue occurred for me when I expected closing to actually just minimize to system tray. I assumed that the unexpected auto-locking behavior would resolve itself if the minimize feature was working. Though I wasn't sure if that's actually how all that is handled internally, so I figured I'd mention its own thing anyways.

> If you refresh the extension options page in Safari, it will show that it's disconnected.

I did try this to see if it would update but it still would show as connected. After I uninstalled the desktop app (before uninstalling the safari extension itself), even then when refreshing the page it thought it was connected, or at least the frontend webpage said it was anyways.

> It will not include KeeWebHttp, this plugin has already been removed from the gallery.

Realized this after downgrading, but I was able to recover a local copy thankfully. Does the KeePassXC extension integration works as a universal stand-in for KeeWebHttp? If not perhaps deprecating the old plugin rather than full removal would be helpful for those on prior versions in the near-term.

> not sure if there would be enough audience for a beta version

Well I guess you've got at least one beta tester here already!
